### PR TITLE
Use IPv6-enabled proxy-init

### PR DESCRIPTION
**Pull request by alpeb**
Followup to linkerd/linkerd2-proxy-init#350

- For the linkerd-control-plane chart added the values.yaml entry `disableIPv6`, defaulting to false. 
- For the linkerd2-cni chart added the entries `iptablesMode` (defaults to "legacy") and `disableIPv6` (defaults to false).

Note this allows routing IPv6 traffic to the proxy by default, but it's just the first step towards IPv6/dual-stack support. More control plane and proxy changes will come next.

### CODE REVIEW

File: .github/workflows/integration.yml [hunk 1]
```diff
@@ -29,7 +29,7 @@ permissions:
 env:
   CARGO_INCREMENTAL: 0
   CARGO_NET_RETRY: 10
-  DOCKER_REGISTRY: ghcr.io/linkerd
+  DOCKER_REGISTRY: ghcr.io/alpeb
```
Comment (olix0r) [inline .github/workflows/integration.yml:32]:
```suggestion
  DOCKER_REGISTRY: ghcr.io/linkerd
```

Comment (olix0r) [reply .github/workflows/integration.yml:32]: > Do not merge yet: We're pulling the images ghcr.io/alpeb/proxy-init:ipv6 and ghcr.io/alpeb/cni-plugin:ipv6 as temporary builds for https://github.com/linkerd/linkerd2-proxy-init/pull/350, while that gets released. For easier review this was isolated in https://github.com/linkerd/linkerd2/commit/b2c819eca46d62b32aa2a7b11db99a5582d97b49 so the real changes are just in https://github.com/linkerd/linkerd2/commit/90319429c1c0c80ff10aad72286b816446d65187

Is this resolved yet?

**Comment by alpeb** [comment]
I've refactored this so that it remains fully compatible the current proxy-init (v2.2.4) and updated the description above accordingly.

File: charts/linkerd2-cni/templates/cni-plugin.yaml [hunk 2]
```diff
@@ -176,7 +177,9 @@ data:
         ],
         {{- end }}
         "simulate": false,
-        "use-wait-flag": {{.Values.useWaitFlag}}
+        "use-wait-flag": {{.Values.useWaitFlag}},
+        "iptables-mode": {{.Values.iptablesMode | quote}},
+        "ipv6": {{ternary "false" "true" .Values.disableIPv6 }}
```
Comment (olix0r) [inline charts/linkerd2-cni/templates/cni-plugin.yaml:182]:
```suggestion
        "ipv6": {{.Values.disableIPv6 | ternary "false" "true"}}
```

Is this clearer? I find the existing syntax pretty obtuse

**Review by olix0r — APPROVED** [review]

File: charts/linkerd2-cni/templates/cni-plugin.yaml (hunk 2 above)
Comment (alpeb) [reply charts/linkerd2-cni/templates/cni-plugin.yaml:182]: Yep, that looks a little better :+1:

**Pull request merged**